### DISCUSSION TITLE
How to get a manifold mesh?

### DISCUSSION THREAD

**Discussion by cshan** [zeux/meshoptimizer#654] [General]
I am using meshoptimizer to do simplification, and then use pmp library to fix holes, but pmp need a manifold mesh, I always got a Complex vertex/Complex edge TopologyException. does meshoptimizer have anyway to generate a manifold mesh?

**Comment by zeux** [comment]
There are no algorithms in meshoptimizer to produce a manifold mesh from a non-manifold mesh.
